core/src/test/scala/org/apache/spark/SparkContextSuite.scala

```diff
@@ -106,6 +106,35 @@ class SparkContextSuite extends FunSuite with LocalSparkContext {
       sc.stop()
     }
   }
+  
+  test("addFile works with relative path") {
```

Comment:
I think it would be good to test both absolute and relative paths.  Maybe we can break the meat of the test out into a separate function and have both tests call it.